pkg/services/multildap/multildap.go

```diff
@@ -92,41 +92,42 @@ func (multiples *MultiLDAP) Login(query *models.LoginUserQuery) (
 	return nil, ErrInvalidCredentials
 }
 
-// User gets a user by login
+// User attempts to find an user by login/username by searching into all of the configured LDAP servers. Then, if the user is found it returns the user alongisde the server it was found.
 func (multiples *MultiLDAP) User(login string) (
 	*models.ExternalUserInfo,
+	*ldap.ServerConfig,
```

Comment:
I'm thinking that it might be a good idea not to return a pointer here. In the current codebase we never re-use the `ldap.ServerConfig` as far as I'm aware, so there shouldn't be any problems. But I could easily see us changing that at some point and then running into problems with a return ServerConfig struct that is modified. I think it's fine to not be defensive while send the config into the LDAP module, you kinda expect whoever is setting up LDAP to not modify the config after its been entered. But I wouldn't neccessarily make the assumptions with config that is returned from LDAP.

Reply:
I agree. The reason why I left it as is due to the place of usage. 

At the moment of writing, the only place where we use the config is for displaying information as part of a short-lived HTTP request and I felt it doesn't really make a difference in there. The chances of a race are very slim IMO.

Reply:
I agree that the risk for problems right now is petty much zero. The problem we could get into is if we ever change it so that we don't re-read the config every time we set up the LDAP services. If we ever start re-using the config we could run into non-obvious problems with this code.

Reply:
Good shout. I have changed the return to a value.